### DISCUSSION TITLE
Any future plans for a Client list Menu?

### DISCUSSION THREAD

**Discussion by davidphilipbarr** [labwc/labwc#986] [Ideas]
Are there any plans to implement a menu comparable to openbox's client-list-combined-menu? I'm really enjoying what's going on with labwc, having root-menus back in my world seems so natural to use. Having a client menu would really make it a stand alone environment.

**Comment by johanmalm** [comment]
It's not been on my TODO list, but I think it'd be fairly straight forward to do.

Having had a quick look at `menu.c` I think we ought to refactor it first before attempting `client-list-combined-menu`. Suggested refactoring: have a `wl_list` for each menu rather than the `struct menu *menus` vector in order to make it easier to free individuals menus (which we would have to do for a client menu every time it's used).
